ql/src/java/org/apache/hadoop/hive/ql/parse/SemanticAnalyzer.java

```diff
@@ -4453,7 +4452,7 @@ private boolean isAggregateInSelect(Node node, Collection<ASTNode> aggregateFunc
    * Returns whether the pattern is a regex expression (instead of a normal
    * string). Normal string is a string with all alphabets/digits and "_".
    */
-  boolean isRegex(String pattern, HiveConf conf) {
+  static boolean isRegex(String pattern, HiveConf conf) {
```

Comment:
This function is called from `SemanticAnalyzer` and its subclasses. I haven't found any other invocations. Is it necessary to change this to `static` ?

Reply:
this patch adds a call to this method from `ParseUtils`  ; that's why it was neccessary to change it to at least `static`

Reply:
found it.